src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -0,0 +1,296 @@
+  }
+  private _indeterminate = false;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(disabled) {
+    this._disabled = coerceBooleanProperty(disabled);
+    this._cdr.markForCheck();
+  }
+  private _disabled = false;
+
+  @Input()
+  get required(): boolean {
+    return this._required;
+  }
+  set required(required) {
+    this._required = coerceBooleanProperty(required);
+  }
+  private _required = false;
+
+  @Output()
+  readonly change: EventEmitter<MatCheckboxChange> = new EventEmitter<MatCheckboxChange>();
+
+  @Output() readonly indeterminateChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  @ViewChild('checkbox') _checkbox: ElementRef<HTMLElement>;
+
+  @ViewChild('nativeCheckbox') _nativeCheckbox: ElementRef<HTMLInputElement>;
+
+  @ViewChild('label') _label: ElementRef<HTMLElement>;
+
+  get inputId(): string {
+    return `${this.id || this._uniqueId}-input`;
+  }
+
+  _checkboxFoundation: MDCCheckboxFoundation;
+
+  _classes: {[key: string]: boolean} = {'mdc-checkbox__native-control': true};
+
+  private _cvaOnChange = (_: boolean) => {};
+
+  private _cvaOnTouch = () => {};
+
+  private _rippleTarget: RippleTarget;
+
+  private _rippleRenderer: RippleRenderer;
+
+  // MDC uses animation events to determine when to update aria-checked which is unreliable.
+  // Therefore we handle it ourselves.
+  private _attrBlacklist = new Set(['aria-checked']);
+
+  private _checkboxAdapter: MDCCheckboxAdapter = {
+    addClass: (className) => this._setClass(className, true),
+    removeClass: (className) => this._setClass(className, false),
+    forceLayout: () => this._platform.isBrowser && this._checkbox.nativeElement.offsetWidth,
+    hasNativeControl: () => !!this._nativeCheckbox.nativeElement,
+    isAttachedToDOM: () => !!this._checkbox.nativeElement.parentNode,
+    isChecked: () => this.checked,
+    isIndeterminate: () => this.indeterminate,
+    removeNativeControlAttr:
+        (attr) => {
+          if (!this._attrBlacklist.has(attr)) {
+            this._nativeCheckbox.nativeElement.removeAttribute(attr);
+          }
+        },
+    setNativeControlAttr:
+        (attr, value) => {
+          if (!this._attrBlacklist.has(attr)) {
+            this._nativeCheckbox.nativeElement.setAttribute(attr, value);
+          }
+        },
+    setNativeControlDisabled: (disabled) => this.disabled = disabled,
+  };
+
+  constructor(
+      private _cdr: ChangeDetectorRef, private _platform: Platform, private _ngZone: NgZone,
+      @Attribute('tabindex') tabIndex: string,
+      @Optional() @Inject(MAT_CHECKBOX_CLICK_ACTION) private _clickAction: MatCheckboxClickAction,
+      @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+    this.tabIndex = parseInt(tabIndex) || 0;
+    this._checkboxFoundation = new MDCCheckboxFoundation(this._checkboxAdapter);
+    // Note: We don't need to set up the MDCCheckboxFoundation. Its only purpose is to manage the
```

Comment:
~~do you mean call foundation.init()? I think you do.~~

**EDIT:** Looks like you do it below.